tools/toolchain/scripts/install_cosma.sh

```diff
@@ -0,0 +1,176 @@
+                case "$FAST_MATH_MODE" in
+                 mkl)
+                 cmake -DCMAKE_INSTALL_PREFIX="${pkg_install_dir}" \
+ 			 -DCOSMA_BLAS=CUDA \
+                         -DCOSMA_SCALAPACK=MKL \
+			 -DCOSMA_WITH_TESTS=NO \
+			 -DCOSMA_WITH_BENCHMARKS=NO \
+			 -DCOSMA_WITH_APPS=NO .. > cmake.log 2>&1
+                     ;;
+                 *)
+                     cmake -DCMAKE_INSTALL_PREFIX="${pkg_install_dir}" \
+                           -DCOSMA_BLAS=CUDA \
+                           -DCOSMA_SCALAPACK=CUSTOM \
+                           -DCOSMA_WITH_TESTS=NO \
+			   -DCOSMA_WITH_BENCHMARKS=NO \
+			   -DCOSMA_WITH_APPS=NO .. > cmake.log 2>&1
+                     ;;
+             esac
+                make -j $NPROCS > make.log 2>&1
+                install -d ${pkg_install_dir}/lib/cuda
+                [ -f libs/grid2grid/src/grid2grid/*.a ] && install -m 644 libs/grid2grid/src/grid2grid/*.a ${pkg_install_dir}/lib/cuda >> install.log 2>&1
+                [ -f libs/options/*.so ] && install -m 644 libs/options/*.so ${pkg_install_dir}/lib/cuda >> install.log 2>&1
+                install -m 644 src/cosma/*.a ${pkg_install_dir}/lib/cuda >> install.log 2>&1
+		[ -f libs/Tiled-MM/src/Tiled-MM/*.a ] && install -m 644 libs/Tiled-MM/src/Tiled-MM/*.a ${pkg_install_dir}/lib/cuda >> install.log 2>&1
+            fi
+        fi
+        COSMA_ROOT="${pkg_install_dir}"
+        COSMA_CFLAGS="-I'${pkg_install_dir}/include'"
+
+        # check if cosma is compiled with 64bits and set up COSMA_LIBDIR accordingly
+        COSMA_LIBDIR="${pkg_install_dir}/lib"
+
+        [ -d  ${pkg_install_dir}/lib64 ] && COSMA_LIBDIR="${pkg_install_dir}/lib64"
+
+        COSMA_LDFLAGS="-L'${COSMA_LIBDIR}' -Wl,-rpath='${COSMA_LIBDIR}'"
+        COSMA_CUDA_LDFLAGS="-L'${COSMA_LIBDIR}/cuda' -Wl,-rpath='${COSMA_LIBDIR}/cuda'"
+        ;;
+    __SYSTEM__)
+        echo "==================== Finding cosma from system paths ===================="
+        check_command pkg-config --modversion cosma
+        add_include_from_paths COSMA_CFLAGS "cosma.h" $INCLUDE_PATHS
+        add_lib_from_paths COSMA_LDFLAGS "libcosma.*" $LIB_PATHS
+        ;;
+    __DONTUSE__)
+        ;;
+    *)
+        echo "==================== Linking cosma to user paths ===================="
+        pkg_install_dir="$with_cosma"
+        check_dir "$pkg_install_dir/lib"
+        check_dir "$pkg_install_dir/lib64"
+        check_dir "$pkg_install_dir/include"
+
+
+        # check if cosma is compiled with 64bits and set up COSMA_LIBDIR accordingly
+        COSMA_LIBDIR="${pkg_install_dir}/lib"
+
+        [ -d  ${pkg_install_dir}/lib64 ] && COSMA_LIBDIR="${pkg_install_dir}/lib64"
+
+        COSMA_CFLAGS="-I'${pkg_install_dir}/include'"
+        COSMA_LDFLAGS="-L'${COSMA_LIBDIR}' -Wl,-rpath='${COSMA_LIBDIR}'"
+        ;;
+esac
+if [ "$with_cosma" != "__DONTUSE__" ] ; then
+	if [ "$ENABLE_CUDA" = "__TRUE__" ] ; then
+    		COSMA_LIBS="-lcosma_pxgemm -lcosma -lgrid2grid -lTiled-MM"
```

Comment:
Here, the `-lTiled-MM` needs to be wrapped in `IF_CUDA` too.

Reply:
IIRC, I stumped over this while trying to build `psmp` with a cuda enabled toolchain.

I believe, the simplest solution would be to just remove lines 134-138 and replace it the definition in from lines 146:

```
COSMA_LIBS="-lcosma_pxgemm -lcosma -lgrid2grid IF_CUDA(-lTiled-MM|)"
```
Also note that all references to `${COSMA_FOO}` are replaced immediately, because they are not escaped. Hence, the lines starting with `export COSMA_` can be remove.